metal/end_model/loss.py

```diff
@@ -20,8 +20,12 @@ class SoftCrossEntropyLoss(nn.Module):
 
     def __init__(self, weight=None, reduction="elementwise_mean"):
         super().__init__()
-        assert weight is None or isinstance(weight, torch.FloatTensor)
-        self.weight = weight
+        self.weight = None
+        if weight is not None:
+            self.weight = torch.FloatTensor(weight)
+            # Register as buffer is standard way to make sure gets moved /
+            # converted with the Module, without making it a Parameter
+            self.register_buffer("class_weights", self.weight)
```

Comment:
@ajratner--nice, register_buffer is much cleaner! But it seems a little confusing to me that in our defaults it's 'loss_weights', in the module it's self.weight, and then we register it as 'class_weights'. I vote we make it 'loss_weights' everywhere.